fix: Generic highlights in types and in method call

MikaelMayer opened the pull request:
* Fixes #68 
Before:
![image](https://user-images.githubusercontent.com/3601079/152874336-435589cd-e613-4a32-9154-e4b8894f911d.png)
After:
![image](https://user-images.githubusercontent.com/3601079/152873680-a257dcf6-f7dd-4618-bde7-440072984aa8.png)

* Fixes #79 
Before:
![image](https://user-images.githubusercontent.com/3601079/152874008-0da8bd2f-4d61-4382-839d-e26de234b5e7.png)
After:
![image](https://user-images.githubusercontent.com/3601079/152874056-08cbd8f7-8b6c-4ab0-b492-983878afe8f8.png)

cpitclaudel commented on syntaxes/Dafny.tmLanguage:
```diff
@@ -54,7 +54,7 @@
 		<dict>
 			<key>begin</key>
-			<string>(\w+)\s*\(</string>
+			<string>(\w+)\s*(&lt;[^&gt;]+&gt;)?\s*\(</string>
```
Can you do this for signatures only?  I don't think it should happen in method bodies.

cpitclaudel replied:
Because in a method body it doesn't work consistently, e.g. `Foo<Bar<Baz>>(xyz)`

MikaelMayer replied:
I fixed it for depths until 3.

keyboardDrummer commented:
Is there a way to write tests for our tmLanguage file ? Seems like it's done here: https://github.com/microsoft/TypeScript-TmLanguage/tree/master/tests

camrein commented:
+1 to the automized tests. We had a past issue with strings, that's why I've removed multi-line string highlighting from normal strings, see #85.

MikaelMayer commented:
@camrein could you have a look again? I updated the description, included a test file for manually reviewing regression tests for now, and fixed one more issue. I just want someone else to have a look at it before merging it, even if I have the rights.

MikaelMayer commented:
I'm going to push more fixes to this branch, because I was able to break the tmLanguage code and find how everything worked together.

MikaelMayer commented:
> Outstanding work @MikaelMayer! Ready to release the new VSCode extension today.

Thanks ! I'm actually terrified with this grammar but I feel better prepared now to handle other syntax highlighting feature requests !
